satpy/readers/hrit_base.py

```diff
@@ -389,11 +324,21 @@ def _read_data_from_disk(self):
         # For reading the image data, unzip_context is faster than generic_open
         dtype, shape = self._get_input_info()
         with utils.unzip_context(self.filename) as fn:
-            with decompressed(fn) if self.compressed else nullcontext(fn) as filename:
-                return np.fromfile(filename,
-                                   offset=self.offset,
-                                   dtype=dtype,
-                                   count=np.prod(shape))
+
+            if self.compressed:
+                return np.frombuffer(
+                    decompress(fn),
+                    offset=self.offset,
+                    dtype=dtype,
+                    count=np.prod(shape)
+                )
+            else:
+                return np.fromfile(
+                    nullcontext(fn),
```

Comment:
```suggestion
                    fn,
```
`nullcontext` isn't needed here, right?